### PR TITLE
Elb prefix changeable

### PR DESCRIPTION
**Pull request by Blubasur**

### CODE REVIEW

File: src/elbwalker.ts [hunk 1]
```diff
@@ -46,8 +47,9 @@ elbwalker.go = function (config: Elbwalker.Config = {}) {
     // custom: use the elbLayer
   }
 
-  // Register all handlers
-  initHandler();
+  if (config.prefix) {
```
Comment (alexanderkirtzel) [inline src/elbwalker.ts]: by setting a default parameter it'll alway be set, but I guess TS isn't happy here.
Also initHandler should always be called to work properly.
Try sth like `const prefix = config.prefix || 'elb'`

File: src/elbwalker.ts [hunk 2]
```diff
@@ -119,6 +121,9 @@ function handleCommand(
   data: Elbwalker.PushData = {},
   elbwalker: Elbwalker.Function,
 ) {
+  if (elbwalker.config.prefix === undefined) {
```
Comment (alexanderkirtzel) [inline src/elbwalker.ts]: if (!elbwalker.config.prefix)

File: src/elbwalker.ts [hunk 3]
```diff
@@ -119,6 +121,9 @@ function handleCommand(
...
+    elbwalker.config.prefix = 'elb';
```
Comment (alexanderkirtzel) [inline src/elbwalker.ts]: const prefix = config.prefix || 'elb'

File: src/elbwalker.ts [hunk 4]
```diff
@@ -166,7 +171,6 @@ function elbLayerInit(elbwalker: Elbwalker.Function) {
 function run(elbwalker: Elbwalker.Function) {
   // When run is called, the walker may start running
   allowRunning = true;
-
```
Comment (alexanderkirtzel) [inline src/elbwalker.ts:169]: nooooo 😱

File: src/elbwalker.ts (hunk 3 above)
Comment (alexanderkirtzel) [inline src/elbwalker.ts]: `const prefix = config.prefix || 'elb'`

File: src/elbwalker.ts [hunk 5]
```diff
@@ -175,15 +179,18 @@ function run(elbwalker: Elbwalker.Function) {
 
   // Load globals properties
   // Due to site performance only once every run
-  globals = getGlobalProperties();
 
+  if (elbwalker.config.prefix) {
```
Comment (alexanderkirtzel) [inline src/elbwalker.ts]: nope

File: src/elbwalker.ts [hunk 6]
```diff
@@ -175,15 +179,18 @@ function run(elbwalker: Elbwalker.Function) {
...
+    globals = getGlobalProperties(elbwalker.config.prefix);
+  }
   // Run predefined elbLayer stack once
   if (firstRun) {
     firstRun = false;
     callPredefined(elbwalker);
   }
-
-  trycatch(triggerLoad)();
+  if (elbwalker.config.prefix) {
```
Comment (alexanderkirtzel) [inline src/elbwalker.ts]: nope

File: src/lib/handler.ts [hunk 7]
```diff
@@ -94,9 +112,16 @@ function observerVisible(duration = 1000): IntersectionObserver | undefined {
 
         if (entry.intersectionRatio >= 0.5) {
           const timer = w.setTimeout(function () {
+            //no good solution
+            if (elbwalker.config.prefix === undefined) {
```
Comment (alexanderkirtzel) [inline src/lib/handler.ts]: Feel free to ask before pr a "no good solution" :D

**Review by alexanderkirtzel — CHANGES_REQUESTED** [review]
Maybe config.prefix should be required to skip the isset part?
Let me know what to update on elb-types

**Comment by Blubasur** [comment]
fixed requested changes

**Review by alexanderkirtzel — APPROVED** [review]
noice

**Pull request merged**